systemtests/systemtestslib/utils.py

```diff
@@ -33,15 +124,19 @@ def get_sym_files(auth_token, url, start_page):
             yield sym_files.pop(0)
         else:
             params["page"] = page
-            resp = requests.get(
+            resp = http_get_with_retry(
                 url,
                 params=params,
                 headers=headers,
-                timeout=CONNECTION_TIMEOUT,
+                timeout=HTTP_DOWNLOAD_TIMEOUT,
             )
             resp.raise_for_status()
             data = resp.json()
-            sym_files = [(record["key"], record["size"]) for record in data["files"]]
+            sym_files = [
+                (record["key"], record["size"])
+                for record in data["files"]
+                if record["completed_at"]
```

Comment:
The file list can come back with files that have records in the table, but _haven't been uploaded_, yet. In these cases, `completed_at` is `None`. We don't want to look at those because if we download them right after generating this list, they're likely to kick up an HTTP 404. We never noticed because the code that downloads these files silently ignored HTTP 404s. Now it calls `resp.raise_for_status()`.